esx_service/vmdk_ops.py

```diff
@@ -837,12 +837,33 @@ def parse_vol_name(full_vol_name):
             raise ValidationError("Volume name contains illegal characters: {0}".format(c))
 
     # Check if the datastore name is too long
-    if ds_name and len(ds_name) > MAX_DS_NAME_LEN:
-        raise ValidationError("Datastore name is too long (max len is {0})".format(MAX_DS_NAME_LEN))
+    if ds_name:
+        if len(ds_name) > MAX_DS_NAME_LEN:
+            raise ValidationError("Datastore name is too long (max len is {0})".format(MAX_DS_NAME_LEN))
+
+        matching_datastores = find_matching_datastores(ds_name)
+
+        # Return error if more than one datastores found
+        if len(matching_datastores) > 1:
+            raise ValidationError("Found multiple datastores with same name (ignoring case difference): {0}".format(matching_datastores))
+
+        # Found exactly one match
+        if len(matching_datastores) == 1:
+            ds_name = matching_datastores[0]
```

Comment:
As @shuklanirdesh82 mentioned, matching_datastores might be empty if the given datastore name does not exist.

I agree the logic here looks a little bit weird, because we should have handled all cases, including len(matching_datastores) == 0. I'm not doing that because that case has already been taken care of in the follow-up validate_datastore() logic.

Ideally the logic here should be incorporated with validate_datastore, but that will need more refactoring, which I'd like to avoid at this point.

Hope this explains your question/confusion @msterin